pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go

```diff
@@ -126,6 +129,16 @@ func (c *SyntheticServiceEntryController) Apply(change *sink.Change) error {
 		return fmt.Errorf("apply: type not supported %s", change.Collection)
 	}
 
+	// Note: run asynchronously to not block mcp receive as it will not return any error
+	c.queue.Push(func() error {
```

Comment:
How does Pilot provide back pressure to Galley? The queue is unbounded. If galley produces faster than pilot can consume, the backlog could get quite large. Pilot would also processes intermediate changes that would immediately be overwritten by latter changes in the queue.

Reply:
This is a normal issue for distributed system.   Message delivery through MQ can not eliminate this problem. If a producer always produces more than the consumer can consume, this is not a healthy system. It should be redesigned from zero.

Reply:
MCP isn't a traditional MQ. It doesn't deliver a sequence of work items. Instead, it delivers the most recent view of a collection of resources. Consumers only need the most recent update. They don't need to reply the entire sequence of events.

Galley doesn't produce more than Pilot can consume. Galley (MCP) accumulates updates and only pushes the latest available resource versions (per-collection). Right now Pilot only requests the next view-of-the-word update when its ready. With this PR, Pilot may accumulate a backlog of updates, only the last of which is actually valid. This is holds true for incremental MCP as well. The delta update is relative to the previous ACK'd update. If Pilot is slow to consume, that delta will grow larger, but it will still be sent as a single update.

Reply:
Thanks for the clarify, make sense for me.